homeassistant/components/frontend/__init__.py

```diff
@@ -211,8 +212,14 @@ def get(self, request, entity_id=None):
 
         panel_url = PANELS[panel]['url'] if panel != 'states' else ''
 
-        # auto login if no password was set
-        no_auth = 'false' if self.hass.config.api.api_password else 'true'
+        no_auth = 'true'
+        if self.hass.config.api.api_password:
+            # require password if set
+            no_auth = 'false'
+            if any(ip_address(request.remote_addr) in trusted_network
```

Comment:
We shouldn't duplicate the check because now it has to be maintained at two places. Better to introduce `HomeAssistantWSGI.is_whitelisted_ip(ip)`